tests/js/HappinessLintBearTest.py

```diff
@@ -17,3 +25,19 @@
 HappinessLintBearTest = verify_local_bear(HappinessLintBear,
                                           valid_files=(good_file,),
                                           invalid_files=(bad_file,))
+
+
+@generate_skip_decorator(HappinessLintBear)
+class HappinessLintBearConfigTest(LocalBearTestHelper):
+
+    USE_SPACE_ERR_MSG = re.escape('"use_spaces=True" is incompatible with '
+                                  'HappinessLintBear, set it to false.')
+
+    def test_validate_use_spaces(self):
+        section = Section('name')
+        section.append(Setting('use_spaces', True))
+        bear = HappinessLintBear(section, Queue())
+
+        with self.assertRaisesRegex(AssertionError, self.USE_SPACE_ERR_MSG):
+            with execute_bear(bear, bad_file) as results:
+                self.assertEqual(''.join(fdict[fname]), good_file)
```

Comment:
please do this, and remove the rest
```
with self.assertRaisesRegex(ValueError, self.USE_SPACE_ERR_MSG):
    bear.execute('test_file', [])
```

Reply:
No error will be raised in this IMO.

Reply:
I mean no assertion error will be raised

Reply:
![image](https://user-images.githubusercontent.com/25884863/33569466-cf144b1e-d94f-11e7-9a86-3c1a6bd83020.png)
If this is what u r telling then it fails the pytest.

Reply:
`execute_bear(bear, 'test_file', [])`*

Reply:
now i am getting confused in what execute_bear does.

Reply:
Yeah a `ValueError` is raised instead

Reply:
`execute_bear` catches any error and reraises an `AssertionError` in place of any other error^^

Reply:
But the changes u asked to make doesnt raise one.

Reply:
Oh it doesn't? Let me see...

Reply:
`execute_bear(bear,"test_file",[])` this is what I have written

Reply:
Yeah and IMO it shouldn't.

Reply:
Ah it seems that inside `execute` a `ValueError` is catched itself, and a logging message gets emitted. So imo it's up to you: Call `bear.run` or instead of `assertRaisesRegex` use `LogCapture` to assert for the log message

Reply:
![image](https://user-images.githubusercontent.com/25884863/33589611-a4fef93c-d99f-11e7-8c8a-cccf0260edd5.png)
If this is what you are telling then -s flag shows "No logging message generated".

Reply:
this execute_bear is something very interesting and tricky ;P

Reply:
please again replace `execute_bear` with `bear.execute` ;)
However, if no logs are generated that's a bit weird. Like said, you can also call `bear.run` directly which is probably easiest.

Reply:
I tried bear .execute as well but mo logs ;P

Reply:
@Makman2 test still fails.

Reply:
> Like said, you can also call bear.run

Reply:
And then don't assert for logs, just for the exception

Reply:
![image](https://user-images.githubusercontent.com/25884863/33617020-aba034d0-da04-11e7-833e-d449651ee1b6.png)
this is what u r telling right?

Reply:
This issue was great to solve, helped me learn a lot.